yt/frontends/fits/data_structures.py

```diff
@@ -410,21 +402,22 @@ def _set_code_unit_attributes(self):
             length_factor = self.wcs.wcs.cdelt[0]
             length_unit = str(file_units[0])
             mylog.info("Found length units of %s." % (length_unit))
-        else:
+        elif "length_unit" not in self.units_override:
```

Comment:
Any chance we'll need a final `else` block?

Reply:
I'll add one. Thanks for catching this.